src/pages/api/profiles/[id].ts

```diff
@@ -69,24 +61,31 @@ async function handleGet(req: NextApiRequest, res: NextApiResponse) {
     }
 }
 
+const allowed = ['first_name', 'last_name', 'website', 'github', 'linkedin', 'twitter', 'biography', 'teamId', 'role']
+
 // PATCH /api/profiles/[id]
 async function handlePatch(req: NextApiRequest, res: NextApiResponse) {
-    if (!(await requireOrgAdmin(req, res))) return
     const profileId = req.query.id as string
+    const session = await getSessionUser(req)
+    if ((await requireOrgAdmin(req, res)) || session?.profileId === profileId) {
+        const params: UpdateProfilePayload = req.body
+        if (!params || Object.keys(params).some((param) => !allowed.includes(param))) return res.status(500)
+        const { teamId, ...other } = params
 
-    const params: UpdateProfilePayload = req.body
-    if (!params.role && !params.teamId) {
-        res.status(400).json({ error: 'Missing params' })
-        return
-    }
+        const data = {
+            ...(teamId ? { team_id: teamId === 'None' ? null : parseInt(teamId) } : {}),
+            ...other,
+        }
 
-    const profile = await prisma.profile.update({
-        where: { id: profileId },
-        data: {
-            ...(params.role ? { role: params?.role } : {}),
-            ...(params.teamId ? { team_id: params?.teamId === 'None' ? null : parseInt(params.teamId) } : {}),
-        },
-    })
+        const profile = await prisma.profile.update({
+            where: { id: profileId },
```

Comment:
```suggestion
            where: { id: session.profileId },
```

Would probably rather default to updating whoever is logged in rather than what the user passed in the URL